main.tf

```diff
@@ -82,8 +82,16 @@ resource "ibm_is_virtual_endpoint_gateway" "vpe" {
     crn           = each.value.service == null ? each.value.crn : local.service_to_endpoint_map[each.value.service]
     resource_type = "provider_cloud_service"
   }
+  depends_on = [time_sleep.wait_30_seconds]
 }
 
+resource "time_sleep" "wait_30_seconds" {
```

Comment:
Perhaps leave a code comment here to say:

Wait 30secs after security group is destroyed before destroying VPE to workaround timing issue which can produce "Target not found" error on destroy